### PR TITLE
Add timeout to calls so they don't pend forever

### PR DESCRIPTION
**Pull request by KillariDev**
this is especially important for retrieving blocks, as interceptor gets stuck in trying to retrieve block while server does not respond

## Summary by CodeRabbit

- **New Features**
  - Search functionality now includes a timeout feature to enhance performance and user experience.
  
- **Enhancements**
  - Improved error logging for failed block retrievals in the Ethereum simulation service.

- **Bug Fixes**
  - Implemented a new fetch request utility with timeout to prevent indefinite waiting during network calls.

### CODE REVIEW

**Comment by KillariDev** [comment]
@coderabbitai review

File: app/ts/simulation/services/EthereumClientService.ts [hunk 1]
```diff
@@ -86,6 +86,7 @@ export class EthereumClientService {
 			this.newBlockAttemptCallback(newBlock, this, gotNewBlock)
 			this.cachedBlock = newBlock
 		} catch(error) {
+			console.log(`Failed to get a block`)
```
Comment (jubalm) [inline app/ts/simulation/services/EthereumClientService.ts:89]: `console.error` maybe more prominent in the logs

**Comment by KillariDev** [comment]
@coderabbitai review

File: app/ts/utils/requests.ts [hunk 2]
```diff
@@ -53,3 +53,11 @@ export const getUniqueRequestIdentifierString = (uniqueRequestIdentifier: Unique
 export const doesUniqueRequestIdentifiersMatch = (a: UniqueRequestIdentifier, b: UniqueRequestIdentifier) => {
 	return a.requestId == b.requestId && a.requestSocket.connectionName === b.requestSocket.connectionName && a.requestSocket.tabId === b.requestSocket.tabId
 }
+
+export async function fetchWithTimeout(resource: RequestInfo | URL, init?: RequestInit | undefined, timeoutS: number = 60000) {
```
Comment (jubalm) [inline app/ts/utils/requests.ts:57]: `timeoutS` may be a typo here with a capital `S`. or you might be referring to milliseconds which in that case could have been `timeoutms`

**Review by jubalm — APPROVED** [review]

File: app/ts/simulation/services/EthereumClientService.ts (hunk 1 above)
Comment (KillariDev) [reply app/ts/simulation/services/EthereumClientService.ts:89]: I don't want to be throwing errors that occur normally and expected at times, as these will show in users "errors" in chrome

**Pull request merged**